inc/classes/CDN/CDN.php

```diff
@@ -36,16 +36,47 @@ public function __construct( Options_Data $options ) {
 	 * @return string
 	 */
 	public function rewrite( $html ) {
-		$pattern = '#(?<url>(?<=[(\"\'])(?:(?:https?:|)' . preg_quote( $this->get_base_url(), '#' ) . ')?\/(?:(?:(?:' . $this->get_allowed_paths() . ')[^\"\')]+)|(?:[^\/\"\'>]+\.[^\/\"\')]+))(?=[\"\')]))#i';
+		$pattern = '#[("\']\s*(?<url>(?:(?:https?:|)' . preg_quote( $this->get_base_url(), '#' ) . ')?\/(?:(?:(?:' . $this->get_allowed_paths() . ')[^"\',)]+)|(?:[^\/"\'>]+\.[^\/"\')]+)))\s*["\')]#i';
 		return preg_replace_callback(
 			$pattern,
 			function( $matches ) {
-				return $this->rewrite_url( $matches['url'] );
+				return str_replace( $matches['url'], $this->rewrite_url( $matches['url'] ), $matches[0] );
 			},
 			$html
 		);
 	}
 
+	/**
+	 * Rewrites URLs in a srcset attribute using the CDN URL
+	 *
+	 * @since 3.4
+	 * @author Remy Perona
+	 *
+	 * @param string $html HTML content.
+	 * @return string
+	 */
+	public function rewrite_srcset( $html ) {
+		$pattern = '#\s+(?:data-lazy-|data-)?srcset\s*=\s*["\']\s*([^"\',\s]+\.[^"\',\s]+(?:\s+\d+(?:w|px|r?em))?(?:\s*,\s*[^"\',\s]+\.[^"\',\s]+\s+\d+(?:w|px|r?em))*)\s*["\']#i';
```

Comment:
`(?:w|px|r?em)`: apparently it should be `[wx]`: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attr-srcset